Mathlib/Algebra/Module/LocalizedModule/Submodule.lean

```diff
@@ -195,4 +195,28 @@ lemma LinearMap.toKerLocalized_isLocalizedModule (g : M →ₗ[R] P) :
     LinearEquiv.ofEq _ _ (localized'_ker_eq_ker_localizedMap S p f f' g)
   IsLocalizedModule.of_linearEquiv p (Submodule.toLocalized' S p f (ker g)) (e.restrictScalars R)
 
+open Submodule IsLocalizedModule
+
+/-- Localization commutes with ranges. -/
+lemma LinearMap.localized'_range_eq_range_localizedMap (g : M →ₗ[R] P) :
+    localized' S p f' (LinearMap.range g) =
+      LinearMap.range ((map p f f' g).extendScalarsOfIsLocalization p S) := by
```

Comment:
I've started adapting new materials in this PR to my #19118 at [here](https://github.com/leanprover-community/mathlib4/pull/19126/commits/8e0bcab8308b5cbe0d39123c5e778a675ed6fb7e).